projects/cli/src/enginematch.cpp

```diff
@@ -165,3 +203,19 @@ void EngineMatch::printRanking()
 {
 	qInfo("%s", qUtf8Printable(m_tournament->results()));
 }
+
+void EngineMatch::onTimer()
```

Comment:
This file existence -based event is a bit weird. So we're checking twice per second if this file exists? What is the use case for this feature?

If we absolutely need this, maybe instead of polling we could use `QFileSystemWatcher` to watch the directory where the file should be?

Reply:
Thank you. Sorry for the late reply. I now replaced the timer-based check and used a `QFileSystemWatcher` and two instances of`QFile`. I separated this code from the rest into a third commit.

Reply:
There are users who are not very comfortable with using signals, so I added the file-based control functionality.